src/time_tracker/db.clj

```diff
@@ -0,0 +1,17 @@
+(ns time-tracker.db
+  (:require [time-tracker.config :as config])
+  (:import com.mchange.v2.c3p0.ComboPooledDataSource))
+
+(defn pool []
+  (let [datasource (doto (ComboPooledDataSource.)
+                     (.setDriverClass "org.postgresql.Driver")
+                     (.setJdbcUrl config/jdbc-uri)
+                     ;; expire excess connections after 30 minutes of inactivity:
+                     (.setMaxIdleTimeExcessConnections (* 30 60))
+                     ;; expire connections after 3 hours of inactivity:
+                     (.setMaxIdleTime (* 3 60 60)))]
+    {:datasource datasource}))
+
+(def pooled-db (delay (pool)))
```

Comment:
It's probably better to explicitly call an `init` function to start your pool, rather than start it at compile/load time. That way you have more control over when you instantiate what in your startup routine.